### PR TITLE
STG1X Functional 

### PR DESCRIPTION
**Pull request by Alireza-Lashkaripour**
Functional added with its Ref.

### CODE REVIEW

**Comment by Alireza-Lashkaripour** [comment]
Konstantin has some comments I will add them.

File: source/dftlib/libxc.F90 [hunk 1]
```diff
@@ -342,6 +342,16 @@ subroutine libxc_input(functional_name, dft_params, tddft_params, functional)
       call functional%add_functional(XC_HYB_GGA_XC_B1LYP, 1.00_fp, hfex=HFEX)
     case ("BHHLYP")
       call functional%add_functional(XC_HYB_GGA_XC_BHANDHLYP, 1.00_fp, hfex=HFEX)
+    case ("STG1X")
+      tddft_params%spc_coco = 0.5_fp
+      tddft_params%spc_ovov = 0.5_fp
+      tddft_params%spc_coov = 0.5_fp
+      call functional%add_functional(XC_HYB_GGA_XC_BHANDHLYP, 1.00_fp, hfex=HFEX) 
+      hfex=0.85
+      write(*,fmt='(3a)') "STG1X = B(0.15)HF(0.85)-LYP functional"
```
Comment (ConstLike) [inline source/dftlib/libxc.F90:351]: `'(3a)'`  -> `'(a)'`

File: source/dftlib/libxc.F90 [hunk 2]
```diff
@@ -342,6 +342,16 @@ subroutine libxc_input(functional_name, dft_params, tddft_params, functional)
       call functional%add_functional(XC_HYB_GGA_XC_B1LYP, 1.00_fp, hfex=HFEX)
     case ("BHHLYP")
       call functional%add_functional(XC_HYB_GGA_XC_BHANDHLYP, 1.00_fp, hfex=HFEX)
+    case ("STG1X")
```
Comment (ConstLike) [inline source/dftlib/libxc.F90:345]: Add test with this functional. MRSF triplet for example.

File: source/dftlib/libxc.F90 [hunk 3]
```diff
@@ -342,6 +342,16 @@ subroutine libxc_input(functional_name, dft_params, tddft_params, functional)
...
+      tddft_params%spc_coco = 0.5_fp
```
Comment (ConstLike) [inline source/dftlib/libxc.F90:346]: Move `tddft_params=...` after `hfex=` before `write()`

**Review by ConstLike — CHANGES_REQUESTED** [review]
Please check my comments

**Comment by Alireza-Lashkaripour** [comment]
Done.

**Review by ConstLike — APPROVED** [review]

**Review by karmachoi — APPROVED** [review]

**Review by karmachoi — APPROVED** [review]

**Pull request merged**